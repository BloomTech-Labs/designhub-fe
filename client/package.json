{
  "name": "client",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@auth0/auth0-spa-js": "^1.2.3",
    "@sentry/browser": "^5.6.3",
    "apexcharts": "^3.8.6",
    "axios": "^0.19.0",
    "dotenv": "^8.1.0",
    "node-sass": "^4.12.0",
    "react": "^16.9.0",
    "react-apexcharts": "^1.3.3",
    "react-calendar-heatmap": "^1.8.1",
    "react-dom": "^16.9.0",
    "react-redux": "^7.1.1",
    "react-router-dom": "^5.0.1",
    "react-scripts": "3.1.1",
    "react-tabs": "^3.0.0",
    "redux": "^4.0.4",
    "redux-logger": "^3.0.6",
    "redux-thunk": "^2.3.0",
    "uuid": "^3.3.3"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test --verbose",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "@testing-library/jest-dom": "^4.1.0",
    "@testing-library/react": "^9.1.4",
    "prettier": "^1.18.2",
    "react-test-renderer": "^16.9.0"
  }
}
