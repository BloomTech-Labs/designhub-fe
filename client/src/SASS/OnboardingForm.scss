@import './palette';
@import './mixins';

.OnboardingForm {
  align-items: baseline;
  background: $black;
  color: $white;
  display: flex;
  flex-direction: column;

  height: 100%;
  width: 100%;

  form {
    @include onboard-form;
  }
}
