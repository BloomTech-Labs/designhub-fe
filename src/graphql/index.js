/*
  QUERIES
*/

export { default as GET_PROJECT_BY_ID_QUERY } from './queries/projects/project';
export { default as GET_ALL_PROJECTS_QUERY } from './queries/projects/projects';
export { default as GET_USER_BY_ID_QUERY } from './queries/users/user';
export { default as GET_ALL_USERS_QUERY } from './queries/users/users';

/*
MUTATIONS
*/

export { default as ADD_COMMENT_MUTATION } from './mutations/comments/addComments';
// export { default as ADD_PHOTO_COMMENT_MUTATION } from './mutations/comments/addPhotoComments';
export { default as ADD_PROJECT_MUTATION } from './mutations/projects/addProject';
export { default as ADD_PROJECT_PHOTO_MUTATION } from './mutations/projects/addProjectPhoto';
export { default as ADD_USER_MUTATION } from './mutations/users/addUser';
// export { default as ADD_CATEGORY_MUTATION } from './mutations/categories/addCategory';
// export { default as ADD_HEATMAP_MUTATION } from './mutations/heatmap/addHeatmap';
// export { default as ADD_INVITE_MUTATION } from './mutations/invites/addInvite';
// export { default as ADD_INVITE_COMMENTS_QUERY } from './mutations/invites/addInviteComments';
// export { default as ADD_INVITE_FOLLOW_MUTATION } from './mutations/invites/addInviteFollow';
// export { default as ADD_INVITE_STARED_MUTATION } from './mutations/invites/addInviteStarred';
export { default as ADD_FOLLOWER_MUTATION } from './mutations/users/addFollower';
// export { default as ADD_STARED_MUTATION } from './mutations/users/addStarred';
// export { default as ADD_USER_RESEARCH_MUTATION } from './mutations/users/addUserResearch';
// export { default as ADD_USER_RESEARCHING_MUTATION } from './mutations/users/addUserResearching';
// export { default as UPDATE_CATEGORY_MUTATION } from './mutations/categories/updateCategory';
// export { default as UPDATE_COMMENT_MUTATION } from './mutations/comments/updateComments';
// export { default as UPDATE_PHOTO_COMMENT_MUTATION } from './mutations/comments/updatePhotoComments';
export { default as UPDATE_PROJECT_MUTATION } from './mutations/projects/updateProject';
export { default as UPDATE_USER_MUTATION } from './mutations/users/updateUser';
// export { default as UPDATE_INVITE_MUTATION } from './mutations/invites/updateInvites';
// export { default as UPDATE_PROJECT_INVITE_MUTATION } from './mutations/invites/updateProjectInvite';
// export { default as UPDATE_PROJECT_INVITES_MUTATION } from './mutations/invites/updateProjectInvites';
export { default as UPDATE_PROJECT_PHOTO_MUTATION } from './mutations/projects/updateProjectPhoto';
// export { default as DELETE_HEATMAP_MUTATION } from './mutations/heatmap/deleteHeatmap';
// export { default as DELETE_PHOTO_COMMENT_MUTATION } from './mutations/comments/deletePhotoComments';
export { default as DELETE_PROJECT_PHOTO_MUTATION } from './mutations/projects/deleteProjectPhoto';
// export { default as DELETE_USER_RESEARCH_MUTATION } from './mutations/users/deleteUserResearch';
// export { default as DELETE_STARED_MUTATION } from './mutations/users/deleteStarred';
// export { default as DELETE_PROJECT_INVITE_MUTATION } from './mutations/invites/deleteProjectInvite';
// export { default as DELETE_CATEGORY_MUTATION } from './mutations/categories/deleteCategory';
export { default as DELETE_FOLLOWER_MUTATION } from './mutations/users/deleteFollower';
export { default as DELETE_PROJECT_MUTATION } from './mutations/projects/deleteProject';
// export { default as DELETE_INVITE_MUTATION } from './mutations/invites/deleteInvite';
export { default as DELETE_COMMENTS_MUTATION } from './mutations/comments/deleteComments';
export { default as DELETE_USER_MUTATION } from './mutations/users/deleteUser';
export { default as SEARCH_MUTATION } from './mutations/users/search';
