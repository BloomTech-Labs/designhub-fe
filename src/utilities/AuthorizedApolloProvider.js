import {
  ApolloClient,
  ApolloProvider,
  createHttpLink,
  InMemoryCache,
} from '@apollo/client';
import { setContext } from '@apollo/link-context';
import React from 'react';

import { useAuth0 } from './auth-spa';

const AuthorizedApolloProvider = ({ children }) => {
  const { getTokenSilently } = useAuth0();

  const httpLink = createHttpLink({
    uri: process.env.REACT_APP_DOMAIN,
  });

  const authLink = setContext(async () => {
    const token = await getTokenSilently();
    return {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    };
  });

  const apolloClient = new ApolloClient({
    link: authLink.concat(httpLink),
    cache: new InMemoryCache(),
    connectToDevTools: true,
  });

  return <ApolloProvider client={apolloClient}>{children}</ApolloProvider>;
};

export default AuthorizedApolloProvider;
