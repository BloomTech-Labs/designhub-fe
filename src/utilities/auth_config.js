const config = {
  domain: `${process.env.REACT_APP_DOMAIN}`,
  clientId: `${process.env.REACT_APP_CLIENT_ID}`,
  audience: `${process.env.REACT_APP_BASE_URL}`,
};

export default config;
