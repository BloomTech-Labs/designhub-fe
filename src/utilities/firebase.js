import * as firebase from "firebase/app";
import "firebase/auth";
import "firebase/firestore";
import "firebase/storage";
// firebase app init and config
const firebaseConfig = {
  apiKey: process.env.REACT_APP_API_KEY,
  authDomain: process.env.REACT_APP_AUTH_DOMAIN,
  databaseURL: process.env.REACT_APP_DATABASE_URL,
  projectId: process.env.REACT_APP_PROJECT_ID,
  storageBucket: process.env.REACT_APP_STORAGE_BUCKET,
  messagingSenderId: process.env.REACT_APP_MESSAGING_SENDER_ID,
  appId: process.env.REACT_APP_APP_ID,
};
firebase.initializeApp(firebaseConfig);
firebase.firestore();
export const storage = firebase.storage();
export default firebase;