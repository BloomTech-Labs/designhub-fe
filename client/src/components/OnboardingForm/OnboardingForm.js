import React, { useEffect, useState } from 'react';

import { useDispatch, useSelector } from 'react-redux';
import { SET_LOGGEDIN, updateUser } from '../../store/actions/usersActions';

import Step1 from './Step1.js';
import Step2 from './Step2.js';

import '../../SASS/OnboardingForm.scss';

const uuidv1 = require('uuid/v1');

const OnboardingForm = () => {
  const dispatch = useDispatch();
  const { currentUser } = useSelector(state => state.users);
  const [formStep, setFormStep] = useState(1);
  const [formUser, setFormUser] = useState({
    avatar: '',
    bio: '',
    email: '',
    firstName: '',
    id: '',
    lastName: '',
    location: '',
    phoneNumber: uuidv1(),
    username: '',
    website: ''
  });

  useEffect(() => {
    let newFormUser = {};
    for (let prop in currentUser) {
      if (currentUser[prop] && formUser[prop] === '')
        newFormUser[prop] = currentUser[prop];
    }
    setFormUser({ ...formUser, ...newFormUser });
    // eslint-disable-next-line
  }, [currentUser]);

  const handleChange = e => {
    setFormUser({ ...formUser, [e.target.name]: e.target.value });
  };

  const handleClick = e => {
    e.preventDefault();
    const move = e.target.name === 'next' ? 'next' : 'prev';
    if (move === 'next' && formStep < stepComponents.length) {
      setFormStep(n => ++n);
    } else if (move === 'prev' && formStep > 1) {
      setFormStep(n => --n);
    }
  };

  const handleSubmit = async (e, id, changes) => {
    e.preventDefault();
    console.log({ id, changes });
    dispatch(updateUser(id, changes));
    dispatch({ type: SET_LOGGEDIN });
  };

  const stepComponents = [Step1, Step2];

  return (
    <div className="OnboardingForm">
      <form onSubmit={handleSubmit}>
        <header>
          <h1>Welcome to DesignHub</h1>
          <h2>Let's get started by creating your profile</h2>
        </header>
        <section className="stepComponents">
          {stepComponents.map((Step, i) => {
            if (i + 1 === formStep) {
              return (
                <Step
                  key={formUser.id}
                  formUser={formUser}
                  onChange={handleChange}
                />
              );
            } else return null;
          })}
        </section>

        <div className="buttons">
          <button name="prev" className="prev-btn" onClick={handleClick}>
            Previous
          </button>
          <button name="next" className="next-btn" onClick={handleClick}>
            Next
          </button>
          <button
            name="next"
            className="next-btn"
            onClick={e => handleSubmit(e, formUser.id, formUser)}
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  );
};

export default OnboardingForm;
