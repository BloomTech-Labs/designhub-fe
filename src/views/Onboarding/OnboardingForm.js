import React, { useState } from 'react';
import { withRouter } from 'react-router-dom';

import { useAuth0 } from '../../utilities/auth-spa.js';
import { storage } from '../../utilities/firebase';
import Loading from '../../common/Loading/index.js';
import Step1 from './Step1.js';
import Step2 from './Step2.js';

import { useMutation, useQuery } from '@apollo/react-hooks';
import {
  UPDATE_USER_MUTATION,
  GET_USER_BY_ID_QUERY,
} from '../../graphql/index';

import './styles.scss';

const OnboardingForm = ({ history, isLoading }) => {
  const [loadingPage, setLoadingPage] = useState(false);

  // user data from auth0-spa
  const { logout, user } = useAuth0();

  const { data } = useQuery(GET_USER_BY_ID_QUERY, {
    variables: { id: user?.sub },
  });

  // individual form steps & state to track which step to display
  const stepComponents = [Step1, Step2];
  const [formStep, setFormStep] = useState(1);

  //conditional render for next/submit/prev buttons
  const submitButton = formStep === stepComponents.length ? true : false;
  const showPrev = formStep === 1 ? false : true;

  //avatar image handler
  const [files, setFiles] = useState([]);

  const [updateUser] = useMutation(UPDATE_USER_MUTATION);

  //local form state populated by auth0 user info
  const [formUser, setFormUser] = useState({
    avatar: data?.user?.avatar || '',
    bio: '',
    firstName: data?.user?.lastName || '',
    id: data?.user?.id,
    lastName: data?.user?.firstName || '',
    location: '',
    username: data?.user?.username || '',
  });

  // alert state for required form inputs
  const [alert, setAlert] = useState({
    username: false,
    firstName: false,
    lastName: false,
  });

  const handleChange = (e, id) => {
    setAlert({
      ...alert,
      [e.target.name]: false,
    });
    setFormUser({ ...formUser, [e.target.name]: e.target.value });
  };

  const handleNextButton = async (e) => {
    e.preventDefault();
    const { username, firstName, lastName } = formUser;
    const newAlert = {};
    if (username.trim().length === 0) newAlert.username = true;
    if (firstName.trim().length === 0) newAlert.firstName = true;
    if (lastName.trim().length === 0) newAlert.lastName = true;
    if (!newAlert.username) {
      try {
        console.log('check userId', data?.user);
        // if (res.data.length !== 0) newAlert.username = 'taken'
      } catch (err) {
        console.error('OnboardingForm.js handleNextButton() ERROR', err);
      }
    }
    const v = Object.values(newAlert);
    if (v.includes(true) || v.includes('taken')) {
      setAlert({ ...alert, ...newAlert });
    } else setFormStep((n) => ++n);
  };

  const handlePrevButton = async (e) => {
    e.preventDefault();
    setFormStep((n) => --n);
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      setLoadingPage(true);
      await handleImageUpload(files);

      updateUser({
        variables: {
          data: {
            id: user?.sub,
            avatar: formUser?.avatar,
            email: user?.email,
            username: formUser?.username,
            firstName: formUser?.firstName,
            lastName: formUser?.lastName,
            bio: formUser?.bio,
            location: formUser?.location,
            website: formUser?.website,
          },
        },
        refetchQueries: [{ query: GET_USER_BY_ID_QUERY }],
      });
      history.push(`/callback`);
    } catch (err) {
      console.error('OnboardingForm.js handleSubmit() ERROR', err);
    }
  };

  const handleImageUpload = async (files) => {
    if (files.length > 0) {
      await files.map(async (file) => {
        try {
          await storage.ref(`/images/${file.name}`).put(file);
          await storage
            .ref('images')
            .child(file.name)
            .getDownloadURL()
            .then(async (firebaseURL) => {
              updateUser({
                variables: {
                  data: {
                    id: user?.sub,
                    avatar: firebaseURL,
                  },
                },
              });
            });
        } catch (err) {
          console.error('OnboardingForm.js handleImageUpload() ERROR', err);
        }
      });
    }
    return data?.avatar;
  };

  if (loadingPage || isLoading) return <Loading />;
  else
    return (
      <>
        <div className="OnboardingForm">
          <form onSubmit={(e) => handleSubmit(e, formUser?.id, formUser)}>
            <section className="stepComponents">
              {stepComponents.map((Step, i) => {
                if (i + 1 === formStep) {
                  return (
                    <Step
                      key={i}
                      alert={alert}
                      files={files}
                      picture={data?.user?.avatar}
                      setFiles={setFiles}
                      formUser={formUser}
                      onChange={handleChange}
                      showPrev={showPrev}
                      handlePrevButton={handlePrevButton}
                      handleNextButton={handleNextButton}
                      logout={logout}
                      submitButton={submitButton}
                    />
                  );
                } else return null;
              })}
            </section>
          </form>
        </div>
      </>
    );
};

export default withRouter(OnboardingForm);
