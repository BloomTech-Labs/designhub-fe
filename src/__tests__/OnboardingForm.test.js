import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import OnboardingForm from '../views/Onboarding/OnboardingForm.js';
import { MockedProvider } from '@apollo/client/testing';
import { ApolloProvider } from '@apollo/react-hooks';
import ApolloClient from 'apollo-boost';
import { useAuth0 } from '../utilities/auth-spa';
import { mount, shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { GET_USER_BY_ID_QUERY, UPDATE_USER_MUTATION } from '../graphql';

configure({ adapter: new Adapter() });
const history = createMemoryHistory();
const user = {
  email: 'johndoe@me.com',
  email_verified: true,
  sub: 'google-oauth2|2147627834623744883746',
};

const client = new ApolloClient({
  uri: process.env.REACT_APP_API_URL,
});

jest.mock('../utilities/auth-spa');

describe('components/OnboardingForm - logged in', () => {
  beforeEach(() => {
    // Mock the Auth0 hook and make it return a logged in state
    useAuth0.mockReturnValue({
      isAuthenticated: true,
      user,
      logout: jest.fn(),
      loginWithRedirect: jest.fn(),
    });
  });
  it('Renders with required props', async () => {
    const wrapper = shallow(
      <ApolloProvider client={client}>
        <Router history={history}>
          <OnboardingForm />
        </Router>
      </ApolloProvider>
    );
    expect(wrapper).toBeTruthy();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

it('submit works', async () => {
  const loadingPage = false;
  const updateUser = {
    user: {
      id: 'google-oauth2|2147627834623744883746',
      firstName: 'Buck',
      lastName: 'bulldog',
      username: 'buckbulldog',
      email: 'buck@bulldog.com',
      location: 'Buckville, USA',
      bio: 'Do not mess with buck',
      website: 'buckwild.org',
      avatar: 'https://bulldog.img',
    },
  };
  const userMocks = [
    {
      request: {
        query: UPDATE_USER_MUTATION,
        variables: {
          id: 'google-oauth2|2147627834623744883746',
        },
      },
      result: {
        data: { updateUser },
      },
    },
  ];
  const component = mount(
    <ApolloProvider client={client}>
      <MockedProvider mocks={userMocks} addTypename={true}>
        <Router history={history}>
          <OnboardingForm id="google-oauth2|2147627834623744883746" loadingPage={loadingPage} />
        </Router>
      </MockedProvider>
    </ApolloProvider>
  );
  
  expect(component.find('form')).toHaveLength(1);
  
  component.find('form').simulate('submit');

});
  