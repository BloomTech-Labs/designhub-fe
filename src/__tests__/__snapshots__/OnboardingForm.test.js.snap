// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/OnboardingForm - logged in Renders with required props 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
